elastic-agent-client.proto

```diff
@@ -112,15 +115,361 @@ message ActionResponse {
     bytes result = 4;
 }
 
+// State codes for the current state.
+enum State {
+    STARTING = 0;
+    CONFIGURING = 1;
+    HEALTHY = 2;
+    DEGRADED = 3;
+    FAILED = 4;
+    STOPPING = 5;
+}
+
+// Type of unit.
+enum UnitType {
+    INPUT = 0;
+    OUTPUT = 1;
+}
+
+// A unit that is part of a collector/shipper.
+message UnitExpected {
+    // Unique ID of the unit.
+    string id = 1;
+    // Unit type.
+    UnitType type = 2;
+    // Expected state of the unit.
+    State state = 3;
+    // Index of the either current configuration or new configuration provided.
+    uint64 config_state_idx = 4;
+    // Resulting configuration. (If the application already has the current `config_state_idx` this
+    // will be empty.)
+    string config = 5;
+}
+
+// A set of units and their expected states and configuration.
+message CheckinExpected {
+    repeated UnitExpected units = 1;
+}
+
+// Observed status for a unit.
+//
+// Contains the currently applied `config_state_idx` (0 in the case of initial start, 1 is the first
+// applied config index) along with the status of the application. In the case that the sent `config_state_idx`
+// doesn't match the expected `config_state_idx` that Elastic Agent expects, the unit is always marked as
+// `CONFIGURING` and a new `UnitExpected` will be sent to so it can have the latest configuration.
+message UnitObserved {
+    // Unique ID of the unit.
+    string id = 1;
+    // Unit type.
+    UnitType type = 2;
+    // Current index of the applied configuration.
+    uint64 config_state_idx = 3;
+    // State of unit.
+    State status = 4;
+    // Nice message for the health status of the unit.
+    // Exposed to users to provide more detail about the status for this single unit.
+    string message = 5;
+    // JSON encoded payload for the status.
+    string payload = 6;
```

Comment:
Any special reason for having here a JSON as string whereas on the other places it's bytes?